packages/api/src/external/commonwell/command/cw-patient-data/create-cw-data.ts

```diff
@@ -0,0 +1,36 @@
+import { NetworkLink } from "@metriport/commonwell-sdk";
+import { executeOnDBTx } from "../../../../models/transaction-wrapper";
+import { CwPatientData, CwPatientDataCreate } from "../../cw-patient-data";
+import { CwPatientDataModel } from "../../models/cw-patient-data";
+import { getCwPatientData } from "./get-cw-data";
+import { updateCwPatientDataWithinDBTx } from "./update-cw-data";
+
+export async function createOrUpdateCwPatientData({
+  id,
+  cxId,
+  cwLinks,
+}: {
+  id: string;
+  cxId: string;
+  cwLinks: NetworkLink[];
+}): Promise<CwPatientData | undefined> {
+  const cwPatientData: CwPatientDataCreate = {
+    id,
+    cxId,
+    data: { links: cwLinks },
+  };
+
+  const updateResult = await executeOnDBTx(CwPatientDataModel.prototype, async transaction => {
+    const existingPatient = await getCwPatientData({
+      id,
+      cxId,
+      transaction,
+    });
+    if (!existingPatient) return undefined;
+    return updateCwPatientDataWithinDBTx(cwPatientData, existingPatient, transaction);
+  });
+  if (updateResult) return updateResult;
+
+  if (!cwLinks.length) return undefined;
```

Comment:
I know this is mimicking CQ's, but I think we either:
- move this to the first line in the function; or
- remove this altogether.

As it stands, we're not creating a new record if it doesn't have any links, but we call an update if there's no links to be added.

_(should prob update CQ's as well)_

Reply:
I removed it as I think this isn't the right place to decide whether or not to save an empty list personally. If I sent in an empty list, I'd expect it to get saved personally...

Reply:
Yeah, its mostly to be consistent and avoid checking for multiple conditions to understand what the state is.

Like, how do we represent "no links"? I'd either choose empty array or undefined/null, but would avoid being able to have both depending on what an external system sends us.

Reply:
Yea, we skip this anyway further up the food chain:
`if (cqLinks.length) await createOrUpdateCQPatientData({ id, cxId, cqLinks });`
`if (cwLinks.length) await createOrUpdateCwPatientData({ id, cxId, cwLinks });`

I think removing it best personally. No code that uses this table should assume the list is non-empty (unless we want to enforce that at the table level).

Reply:
Once we switch to a historical version of the table, noted here https://github.com/metriport/metriport-internal/issues/1844 I think we should actually remove the empty checks anyway, as the idea is to always save the result even if empty so that we have a full history.

Reply:
@leite08 please resolve if you agree with removing it. We check any before this, so removing it will have no affect anyway :).